source/configuration/modules/omfile.rst

```diff
@@ -1,434 +1,441 @@
-   the rest of the engine to lossless operations.
+If different owners or groups are specified for new files or directories, and rsyslogd fails to set them:
 
-.. function::  sig.provider [providerName]
+* If set to *on*, it will log an error and NOT write to the file.
+* If it is set to *off*, the error will be ignored and processing continues.
+  Keep in mind that in the event of an error, files may be (in)accessible due to permissions.
 
-   *Default: no signature provider*
+===========  ===========
+createDirs:  "on|off"
+Default:     "on"
+Legacy:      $CreateDirs
+===========  ===========
 
-   Selects a signature provider for log signing. By selecting a provider,
-   the signature feature is turned on.
+Create directories on an as-needed basis
 
-   Currently, there are two providers available ":doc:`gt <sigprov_gt>`" and ":doc:`ksi <sigprov_ksi>`". 
+Expert parameters
+-----------------
 
-.. function::  cry.provider [providerName]
+**
+If you're not familiar with rsyslog internals, that is a good indication that you should NOT use these parameters.
 
-   *Default: no crypto provider*
+They almost never need to be changed, even on high load systems, so benchmarks should be run before and after changing any of them because sometimes they could lead to undesired non-intuitive performance impact.
+**
 
-   Selects a crypto provider for log encryption. By selecting a provider,
-   the encryption feature is turned on.
+========  ========
+sync:     "on|off"
+Default:  "off"
+========  ========
 
-   Currently, there only is one provider called ":doc:`gcry <../cryprov_gcry>`".
+Enables file syncing capability.
 
-See Also
---------
+When enabled, rsyslog does a sync to the data file as well as the
+directory it resides after processing each batch. There currently
+is no way to sync only after each n-th batch.
 
-- `Sign log messages through signature provider
-  Guardtime <http://www.rsyslog.com/how-to-sign-log-messages-through-signature-provider-guardtime/>`_
+**Enabling sync causes a severe performance hit**.
+Actually, it slows omfile so much down, that the probability of loosing messages increases.
 
-Caveats/Known Bugs
-------------------
+In short, you should enable syncing only if you know exactly what you do, and fully understand how the rest of the engine works, and have tuned the rest of the engine to lossless operations.
 
--  people often report problems that dynafiles are not properly created.
-   The common cause for this problem is SELinux rules, which do not permit
-   the create of those files (check generated file names and pathes!). The
-   same happens for generic permission issues (this is often a problem
-   under Ubuntu where permissions are dropped by default)
+Log signing and encryption
+--------------------------
 
--  One needs to be careful with log rotation if signatures and/or
+**TODO** One needs to be careful with log rotation if signatures and/or
```

Comment:
IE: Because I didn't review that part and don't understand how it works yet (IMHO mandatory for properly document it)